docs/CartRestApiDoc.md

```diff
@@ -0,0 +1,514 @@
+| shoppingSessionId                 | string    | true     | The identifier of the shopping session (the cart info)   |
+| itemsQuantity      | integer   | true     | The quantity of items          |
+| totalPrice         | number    | true     | The total price of all items |
+| items           | item list | true     | The item list                |
+
+### Item
+
+| Parameter          | Type      | Required | Description                   |
+|--------------------|-----------|----------|-------------------------------|
+| shoppingSessionItemId                 | integer   | true     | The identifier of the item    |
+| productsQuantity   | integer   | true     | The quantity of products      |
+| totalProductsPrice | integer   | true     | The total products price      |
+| productInfo          | integer   | true     | The product description       |
+
+**Response Example:**
+
+```json
+{
+    "shoppingSessionId": "fddfgd-fdgdfgdfh-hdfhdfh-436346dfhd-hdfhdf",
+    "itemsQuantity": "2",
+    "totalPrice": "3200",
+    "items": [
+        {
+            "shoppingSessionItemId": "463463gfd-436fdhtgery3-hdhgdfghdgh46-hdhdghd",
+            "productsQuantity": "2",
+            "totalProductsPrice": "1500",
+            "productInfo": {
+                "productData": "productData"
+            }
+        },
+        {
+            "shoppingSessionItemId": "463463gfd-436fdhtgery3-hdhgdfghdgh46-hdhdghd",
+            "productsQuantity": "3",
+            "totalProductsPrice": "1700",
+            "productInfo": {
+                "productData": "productData"
+            }
+        }
+    ]
+}
+```
+
+**400 Bad Request**
+
+If the client sends an invalid request, a 400 Bad Request should be returned.
+
+| Parameter | Type      | Required | Description                                         |
+|-----------|-----------|----------|-----------------------------------------------------|
+| message | string    | true | Error message indicating the reason for the failure. |
+| timestamp | timestamp | true | Indicating the failure time.                        |
+| code | integer   | true | Error code.    |
+
+**Response Examples:**
+
+```json
+{
+    "message": "The 'shoppingSessionId' cannot be empty",
+    "timestamp": "2023-06-04 18:24:54",
+    "code": 12 // TODO Create errorCode hierachy
+}
+```
+
+```json
+{
+    "message": "Invalid 'shoppingSessionId' value",
+    "timestamp": "2023-06-04 18:24:54",
+    "code": 14 // TODO Create errorCode hierachy
+}
+```
+
+```json
+{
+    "message": "The 'itemId' cannot be empty",
+    "timestamp": "2023-06-04 18:24:54",
+    "code": 15 // TODO Create errorCode hierachy
+}
+```
+
+```json
+{
+    "message": "Invalid 'itemId' value",
+    "timestamp": "2023-06-04 18:24:54",
+    "code": 16 // TODO Create errorCode hierachy
+}
+```
+
+------------------------------------------------------------------------------
+
+## 3. Remove the item in the cart
+
+### HTTP DELETE .../api/v1/cart/{shoppingSessionId}/items/
```

Comment:
I would add .../api/v1/cart/{shoppingSessionId}/items/{shoppingSessionItemId}.
If we talking about "THE item".